chef/cookbooks/crowbar/recipes/prepare-upgrade-scripts.rb

```diff
@@ -113,6 +113,12 @@
 
 nova = search(:node, "run_list_map:nova-controller").first
 
+unless nova.nil?
+  keystone_settings = KeystoneHelper.keystone_settings(nova, "nova")
```

Comment:
Hm, AFAIK this does a chef search inside to find the keystone node (```roles:keystone-server AND keystone_config_environment:keystone-config-#{instance}")```, this doesn't work in the current stated of upgrade, does it? (If that chef-search does yield anything it will try to fall back to the node it got passed, ```nova``` in your case, which might be wrong